owl/list.scm

```diff
@@ -180,6 +179,7 @@
 
       ;; misc
 
+      ;; list -> element | #f, SRFI-1
```

Comment:
The current plan related to documentation is to have some fixed text and some content mined from the sources. `(owl proof)` library checks some examples while loading a library, which works as a simple error check, but also provides a way to get reliable examples about usage of functions. The current idea is for `make manual.pdf` to process libraries by including the initial comment section as the main description, and have an index of all exported values along with information from comments like this and examples in which the definition occurs from the library source.